test/string-checker.js

```diff
@@ -207,6 +207,32 @@ describe('modules/string-checker', function() {
         });
     });
 
+    describe('esprima options', function() {
+        var customDescription = 'in no way a real error message';
+        var customEsprima = {
+            parse: function(code, options) {
+                var error = new Error();
+                error.description = options.foobar === 'qux' ? customDescription : 'should never happen';
+                error.lineNumber = 1;
+                error.column = 0;
+
+                throw error;
+            }
+        };
+
+        it('uses custom esprima options when set in the config', function() {
+            checker = new Checker({ esprima: customEsprima });
```

Comment:
before this, Add `var sinon = require('sinon');` to the top of this module, then: `sinon.spy(customEsprima, 'parse');`. Also, don't pass `esprima` into the construction, pass it in via `checker.configure`, which is the preferred way.